lfs.c

```diff
@@ -4803,10 +4804,10 @@ static int lfs_fs_pred(lfs_t *lfs,
         .i = 1,
         .period = 1,
     };
-    int err = LFS_ERR_OK;
+    int err1 = LFS_ERR_OK;
     while (!lfs_pair_isnull(pdir->tail)) {
-        err = lfs_tortoise_detectcycles(pdir, &tortoise);
-        if (err < 0) {
+        err1 = lfs_tortoise_detectcycles(pdir, &tortoise);
+        if (err1 < 0) {
```

Comment:
Inlining this check would eliminate the need for `err1` entirely AFAICS.
```suggestion
        if (lfs_tortoise_detectcycles(pdir, &tortoise) < 0) {
```